version: "3.7"
services:

## Syncthing â€“ Continuous File Synchronization

  ## Fix inotify limit error - Need to run on host
  #https://docs.syncthing.net/users/faq.html#inotify-limits
  # echo "fs.inotify.max_user_watches=204800" | sudo tee -a /etc/sysctl.d/90-override.conf
  # This only takes effect after a reboot. To adjust the limit immediately, run:
  # sudo sh -c 'echo 204800 > /proc/sys/fs/inotify/max_user_watches'

  syncthing:
    container_name: syncthing
    image: linuxserver/syncthing:latest
    restart: always
    networks:
      home_net:
        ipv4_address: 192.168.0.77
    ports:
#      - "8384:8384" # Application WebUI
      - "22000:22000" # Listening port
      - "21027:21027/udp" # Protocol discovery
    security_opt:
      - no-new-privileges:true
    volumes:
      - $VOLUME/code:/data/code
      - $VOLUME/sync:/data/default
      - $USERDIR/syncthing:/config
    environment:
      - PUID=$PUID
      - PGID=$PGID
      - TZ=$TZ
      - UMASK_SET=022
    labels:
      - "traefik.enable=true"
      ## HTTP Routers
      - "traefik.http.routers.syncthing-rtr.entrypoints=https"
      - "traefik.http.routers.syncthing-rtr.rule=HostHeader(`sync.$DOMAINNAME`)"
      - "traefik.http.routers.syncthing-rtr.tls=true"
      ## Middlewares
      - "traefik.http.routers.syncthing-rtr.middlewares=chain-authelia@file"
      ## HTTP Services
      - "traefik.http.routers.syncthing-rtr.service=syncthing-svc"
      - "traefik.http.services.syncthing-svc.loadbalancer.server.port=8384"
networks:
  home_net:
    external: true

